OneCommunityGlobal/HighestGoodNetworkApp
# Vishala circleci reduce load

# Description
This PR reduces the React client build time by reducing bundle size. Previously resource class was increased to large size to fix the repeated circleci build error. This PR intends to reduce the initial load of the app and reduce the bundle size.


## Main changes explained:
- Added some caching steps on circleci/config.yml to not download the dependencies on every build , rather just download when package-lock.json is touched.
- Removed some unused dependencies from package.json
- Implemented code splitting to reduce the bundle size

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally

**3. Code splitting has been implemented on the below components**
Projects 
WeeklySummariesReport
Reports 
PeopleReport 
ProjectReport 
TeamReport 
Members 
WBS 
WBSDetail
SingleTask 
SameFolderTasks 
UserManagement 
UserProfile 
BadgeManagement 
PermissionsManagement 
UserRoleTab 
Teams 

**4.Removed the below unused dependencies.**
* @uiw/react-heat-map
* @uiw/react-tooltip
* acorn
* jest-mock-axios
* js-levenshtein
* nanoid
* react-calendar
* react-date-picker
* react-image
* react-moment
* react-router-bootstrap
* websocket-extensions
* fetch-mock
* node-fetch
* node-sass
* sinon
* surge
5. Please test all the functionalities that you know of in relation to the mentioned components and please do check if the mentioned dependencies are not used anywhere.
